### PR TITLE
feat: add avatar click event

### PR DESCRIPTION
**Pull request by pbosio**

### CODE REVIEW

**Review by leanmendoza — APPROVED** [review]

File: src/index.d.ts [hunk 1]
```diff
@@ -519,4 +519,14 @@ export interface IEvents {
     serverName: string
     displayName: string
   }
+
+  /** Triggered when other player's avatar is clicked */
+  playerClicked: {
+    userId: string
+    ray: {
+      origin: number,
+      direction: number,
```
Comment (gonpombo8) [inline src/index.d.ts]: remove trailing comma 😄

**Review by gonpombo8 — APPROVED** [review]

File: posix.api.md [hunk 2]
```diff
@@ -217,6 +217,15 @@ export interface IEvents {
     text: string
   }
 
+  playerClicked: {
+    userId: string
+    ray: {
+      origin: number
```
Comment (menduz) [inline posix.api.md]: Should origin and direction be vector3?

**Review by menduz — CHANGES_REQUESTED** [review]

File: posix.api.md (hunk 2 above)
Comment (pbosio) [reply posix.api.md]: 🤦‍♂️ Yes! sorry, thanks! fixed :)

**Review by menduz — APPROVED** [review]
:shipit:

**Comment by menduz** [comment]
One last question before merging: should we include the hit point? If we decide to add it later then it will become optional due to backwards compatibility reasons (first version won't have it)

**Comment by pbosio** [comment]
> One last question before merging: should we include the hit point? If we decide to add it later then it will become optional due to backwards compatibility reasons (first version won't have it)

I though about that for a while.
We have for input's hit:
```
/** Length of the ray */
length: number
/** If the ray hits a mesh the intersection point will be this */
hitPoint: ReadOnlyVector3
/** If the mesh has a name, it will be assigned to meshName */
meshName: string
/** Normal of the hit */
normal: ReadOnlyVector3
/** Normal of the hit, in world space */
worldNormal: ReadOnlyVector3
/** Hit entity ID if any */
entityId: EntityID
```
For this new event for clicking avatars:
`length` I added as `distance` in the `ray` field
`hitPoint` can be easily calculated with ray's `origin` + `direction` * `distance` 
`meshName` I don't think is really relevant for avatars since it won't be a mesh added by the developer
`normal` and `worldNormal` might be worth adding it but I can only think on super specific use cases 
`entityId` I can't currently think of a good use for that

So I think that for the time being we are exposing the most relevant information and I didn't want to increase the size of the message just because "just in case" :P

**Pull request merged**